Derivco.Orniscient/Derivco.Orniscient.Proxy/OrniscientLinkMap.cs

```diff
@@ -19,27 +19,38 @@ private OrniscientLinkMap()
             }
         }
 
-        
+
         private void CreateTypeMap()
         {
             _typeMap = new Dictionary<Type, OrniscientGrain>();
-            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies()) 
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
             {
-                foreach (var type in assembly.GetTypes())
+                try
                 {
-                    //if this type is not a grain we do not want it..
-                    if (!typeof(IGrain).IsAssignableFrom(type))
-                        continue;
+                    foreach (var type in assembly.GetTypes())
+                    {
+
+                        //if this type is not a grain we do not want it..
+                        if (!typeof(IGrain).IsAssignableFrom(type))
+                            continue;
+
+                        var attribs = type.GetCustomAttributes(typeof(Attributes.OrniscientGrain), false);
+                        var orniscientInfo = attribs.FirstOrDefault() as OrniscientGrain ?? new OrniscientGrain();
+                        orniscientInfo.IdentityType = GetIdentityType(type);
+
+                        if (orniscientInfo.HasLinkFromType && string.IsNullOrEmpty(orniscientInfo.DefaultLinkFromTypeId))
+                        {
+                            orniscientInfo.DefaultLinkFromTypeId = GetDefaultLinkFromTypeId(type);
+                        }
+                        _typeMap.Add(type, orniscientInfo);
 
-                    var attribs = type.GetCustomAttributes(typeof(Attributes.OrniscientGrain), false);
-                    var orniscientInfo = attribs.FirstOrDefault() as OrniscientGrain ?? new OrniscientGrain();
-                    orniscientInfo.IdentityType = GetIdentityType(type);
 
-                    if (orniscientInfo.HasLinkFromType && string.IsNullOrEmpty(orniscientInfo.DefaultLinkFromTypeId))
-                    {
-                        orniscientInfo.DefaultLinkFromTypeId = GetDefaultLinkFromTypeId(type);
                     }
-                    _typeMap.Add(type, orniscientInfo);
+                }
+                catch (Exception)
+                {
```

Comment:
If this is expected behaviour, can we log a partial failure? Conscious if we had trouble parsing an assembly it would be extremely difficulty to identify when there's no feedback and we continue execution.